crypto_automations/automation/portfolio.py

```diff
@@ -0,0 +1,38 @@
+#  This file is part of Crypto-Automations (https://github.com/Drakkar-Software/Crypto-Automations)
+#  Copyright (c) 2023 Drakkar-Software, All rights reserved.
+#
+#  OctoBot is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU General Public License
+#  as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  OctoBot is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public
+#  License along with Crypto-Automations. If not, see <https://www.gnu.org/licenses/>.
+import asyncio
+
+from crypto_automations.models.automation import Automation
+
+
+class Portfolio(Automation):
+    def __init__(self, exchange, refresh_delay=30, on_refresh_async_callback=None):
+        super().__init__()
+        self.exchange = exchange
+        self.refresh_delay = refresh_delay
+        self.on_refresh_async_callback = on_refresh_async_callback
+
+        self.portfolio = None
+
+    async def run(self):
+        # refresh portfolio based on refresh_delay
+
+        # TODO fetch pf on exchange
+
+        await asyncio.sleep(self.refresh_delay)
```

Comment:
Here we should only sleep if self.refresh_delay is defined to avoid allowing an async loop iteration (which can create side effects)

Reply:
right